src/MSRIOGroup.cpp

```diff
@@ -1176,6 +1288,15 @@ namespace geopm
                     description = field_data["description"].string_value();
                 }
 
+                if (msr_field_name == "QM_EVT_SEL:RMID" || msr_field_name == "PQR_ASSOC:RMID") {
+                    if((end_bit - begin_bit) != m_rdt.rmid_bit_width) {
+                        throw Exception("MSRIOGroup::" + std::string(__func__) + "(): RMID bit width " +
+                                        std::to_string(end_bit - begin_bit) + "  does not match CPUID value: " +
+                                        std::to_string(m_rdt.rmid_bit_width), GEOPM_ERROR_INVALID, __FILE__,
+                                        __LINE__);
+                    }
+                }
+
```

Comment:
Special casing in JSON parse because we can't easily access the signal bit fields later to check if they're valid.  This also is the earliest point at which we can catch a RMID max val vs bit field mismatch.